subprojects/core-api/src/main/java/org/gradle/process/ExecSpec.java

```diff
@@ -15,35 +15,25 @@
  */
 package org.gradle.process;
 
-import org.gradle.internal.instrumentation.api.annotations.ToBeReplacedByLazyProperty;
+import org.gradle.api.provider.ListProperty;
+import org.gradle.api.provider.Provider;
+import org.gradle.internal.instrumentation.api.annotations.ReplacedAccessor;
+import org.gradle.internal.instrumentation.api.annotations.ReplacedAccessor.AccessorType;
+import org.gradle.internal.instrumentation.api.annotations.ReplacesEagerProperty;
 
 import java.util.List;
 
 /**
  * Specified the options for executing some command.
  */
 public interface ExecSpec extends BaseExecSpec {
-    /**
-     * Sets the full command line, including the executable to be executed plus its arguments.
-     *
-     * @param args the command plus the args to be executed
-     * @since 4.0
-     */
-    void setCommandLine(List<String> args);
 
     /**
-     * Sets the full command line, including the executable to be executed plus its arguments.
-     *
-     * @param args the command plus the args to be executed
-     */
-    void setCommandLine(Object... args);
-
-    /**
-     * Sets the full command line, including the executable to be executed plus its arguments.
-     *
-     * @param args the command plus the args to be executed
+     * {@inheritDoc}
      */
-    void setCommandLine(Iterable<?> args);
+    @Override
+    @ReplacesEagerProperty(adapter = ExecSpecAdapters.CommandLineAdapter.class)
+    Provider<List<String>> getCommandLine();
```

Comment:
💭 CommandLine is now a Provider<List<String>> since it's actually a calculated value from executable and args. To set full command line you have to use `commandLine(...)` methods, `setCommandLine(...)` is removed. Also you can't use `commandLine = <>`  anymore